bitcash/network/APIs/BitcoinDotComAPI.py

```diff
@@ -0,0 +1,132 @@
+import requests
+from decimal import Decimal
+from bitcash.exceptions import InvalidEndpointURLProvided
+from bitcash.network import currency_to_satoshi
+from bitcash.network.meta import Unspent
+from bitcash.network.transaction import Transaction, TxPart
+
+# This class is the interface for Bitcash to interact with
+# Bitcoin.com based RESTful interfaces.
+
+BCH_TO_SAT_MULTIPLIER = 100000000
+# TODO: Refactor constant above into a 'constants.py' file
+
+
+class BitcoinDotComAPI:
+    """ rest.bitcoin.com API """
+
+    def __init__(self, network_endpoint: str):
+        try:
+            assert isinstance(network_endpoint, str)
+            assert network_endpoint[:4] == "http"
+            assert network_endpoint[-4:] == "/v2/"
+        except AssertionError:
+            raise InvalidEndpointURLProvided(f"Provided endpoint '{network_endpoint}' is not a valid URL for a Bitcoin.com-based REST endpoint")
+
+        self.network_endpoint = network_endpoint
+
+    # Default endpoints to use for this interface
+    DEFAULT_ENDPOINTS = {
+        "mainnet": ["https://rest.bch.actorforth.org/v2/",
+                    "https://rest.bitcoin.com/v2/"],
+        "testnet": ["https://trest.bitcoin.com/v2/"],
+        "regtest": ["http://localhost:12500/v2/"],
+    }
+
+    # Paths specific to rest.bitcoin.com-based endpoints
+    PATHS = {
+        "unspent": "address/utxo/{}",
+        "address": "address/details/{}",
+        "raw-tx": "rawtransactions/sendRawTransaction/{}",
+        "tx-details": "transaction/details/{}",
+    }
+
+    @classmethod
+    def get_default_endpoints(cls, network):
```

Comment:
Good opportunity for typing here. network is a str, I presume.

Reply:
And returns a str? `-> str`.

Reply:
I'm also a fan of types, and there's plenty of opportunities in this code base to add type hinting .. but it's a bit out of the scope of this specific PR. Perhaps a better fit for a future 'refactor' type PR?

Network is a str, yes. That specific function returns a list (of str) though, so -> list

Reply:
`from typing import List`, `-> List[str]`

Could definitely wait for another PR.